client/webserver/site/src/js/app.js

```diff
@@ -468,17 +468,28 @@ export default class Application {
         const order = note.order
         const mkt = this.user.exchanges[order.host].markets[order.market]
         if (mkt.orders) {
+          let found = false
           for (const i in mkt.orders) {
             if (mkt.orders[i].id === order.id) {
               mkt.orders[i] = order
+              found = true
               break
             }
           }
+          // If notifaction order isn't part of the market orders list
+          // we add the order to list manually as it means this order was
+          // just placed.
+          if (!found) mkt.orders = [...mkt.orders, order]
+        } else {
+          // If user obj has no orders this also means order was just placed, we
+          // add it manually.
+          mkt.order = [order]
```

Comment:
Consider writing a helper function.

```
const updateOrder = (mkt, ord) => {
  for (const i in mkt.orders || []) {
    if (mkt.orders[i].id === ord.id) {
      mkt.orders[i] = ord
      return true
    }
  }
  return false
}
```
so that you can just

```
if (!mkt.orders) mkt.orders = [order]
else if (!updateOrder(mkt, order)) mkt.orders.push(order)
```

Reply:
👍